### PR TITLE
start documenting system.h with doxygen

### PR DESCRIPTION
**Pull request by edg-l**
work in progress

### CODE REVIEW

File: src/base/system.h [hunk 1]
```diff
@@ -32,21 +32,24 @@
 extern "C" {
 #endif
 
-/* Group: Debug */
-/*
-	Function: dbg_assert
-		Breaks into the debugger based on a test.
-
-	Parameters:
-		test - Result of the test.
```
Comment (def-) [inline src/base/system.h:41]: Is it possible to instruct Doxygen to use a format closer to our existing one?

Comment (Jupeyy) [reply src/base/system.h:41]: did we ever use our existing one? xd

Comment (def-) [reply src/base/system.h:41]: I just want to keep the diff as small as possible.

Comment (Jupeyy) [reply src/base/system.h:41]: I guess for the purpose of switching to a useful documenting style it's worth to change the syntax.
We already use this in every place where we documented stuff (component.h, ui.h)

Comment (edg-l) [reply src/base/system.h:41]: > Is it possible to instruct Doxygen to use a format closer to our existing one?

no

**Comment by heinrich5991** [comment]
Can you resolve the merge conflicts so we can merge it?

**Comment by edg-l** [comment]
it's not complete, but i guess it can be merged

**Pull request merged**